build_tools/cmake/build_and_test_asan.sh

```diff
@@ -32,6 +38,11 @@ CMAKE_ARGS=(
   "-DPython3_EXECUTABLE=${IREE_PYTHON3_EXECUTABLE}"
   "-DPYTHON_EXECUTABLE=${IREE_PYTHON3_EXECUTABLE}"
   "-DIREE_ENABLE_ASSERTIONS=${IREE_ENABLE_ASSERTIONS}"
+
+  "-DIREE_HAL_DRIVER_CUDA=${IREE_HAL_DRIVER_CUDA}"
+  "-DIREE_HAL_DRIVER_HIP=${IREE_HAL_DRIVER_HIP}"
```

Comment:
We aren't testing ASan on GPU machines on CI right now, so some of this _could_ be left off. Anyone running locally is likely not using these scripts.

Probably fine to keep for consistency/completeness though.